Lib/typing.py

```diff
@@ -2822,7 +2821,7 @@ def __subclasscheck__(cls, other):
     __instancecheck__ = __subclasscheck__
 
 
-def TypedDict(typename, fields=None, /, *, total=True, **kwargs):
+def TypedDict(typename, fields=None, /, *, total=True):
```

Comment:
If I apply this diff, no tests fail, and we have a cleaner signature. I think the only reason why we had a `None` default for `fields` was to support the kwargs-based syntax:

```diff
diff --git a/Lib/typing.py b/Lib/typing.py
index 13f0883e3b..939ced78cb 100644
--- a/Lib/typing.py
+++ b/Lib/typing.py
@@ -2821,7 +2821,7 @@ def __subclasscheck__(cls, other):
     __instancecheck__ = __subclasscheck__


-def TypedDict(typename, fields=None, /, *, total=True):
+def TypedDict(typename, fields, /, *, total=True):
     """A simple typed namespace. At runtime it is equivalent to a plain dict.

     TypedDict creates a dictionary type that expects all of its
@@ -2859,9 +2859,6 @@ class point2D(TypedDict, total=False):
     the total argument. True is the default, and makes all items defined in the
     class body be required.
     """
-    if fields is None:
-        fields = {}
-
     ns = {'__annotations__': dict(fields)}
     module = _caller()
     if module is not None:
```

_Strictly_ speaking, this would break backwards compatibility, in that it was previously okay to do this, and we haven't emitted deprecation warnings telling people that they can't:

```pycon
>>> import typing
>>> X = typing.TypedDict('X', None)
```

It feels pretty unlikely that anybody _would_ have been doing that, though, so I _think_ we should be okay to get rid of the `=None` default. @JelleZijlstra what do you think?

Reply:
I also noticed this and didn't mention it on the PR because empty TypedDicts probably are occasionally useful, so we shouldn't unnecessarily break them. So I'd rather leave this unchanged.

Reply:
You could still create an empty `TypedDict` just by doing `X = TypedDict("X", {})`. But I guess your concern is that we'd be disallowing `X = TypedDict("X")`?

Reply:
Yes

Reply:
Mypy does not support `X = TypedDict("X")` FWIW: https://mypy-play.net/?mypy=latest&python=3.11&gist=c8ead0c457f9fbdc3bed056c4248d06e

Reply:
Nor does pyright. (Pyre doesn't seem to support the call-based syntax at all.)

Reply:
Still, if it worked at runtime and we didn't emit DeprecationWarnings we shouldn't remove it.

Reply:
Alright, alright :)